Position Score above fold

dfdumaresq opened the pull request:
As a newbie to the AIA, I found I was compelled to constantly scroll down to see how my choices had affected my Assessment Score. So I've moved the score panel to the middle-right of the screen.
The results are I can now see that my score can also influence the Impact level.
![image](https://user-images.githubusercontent.com/20051/62815975-4fbb8300-bad5-11e9-8cb2-64612d846ead.png)

After adding Health Related services
![image](https://user-images.githubusercontent.com/20051/62815986-77aae680-bad5-11e9-8ade-b5c3363268ee.png)

gcharest commented:
Oh, that's nice! I'm not in the office at the moment but will try to find a
minute to review!

Le ven. 9 août 2019 21 h 42, dfdumaresq <notifications@github.com> a écrit :

> As a newbie to the AIA, I found I was compelled to constantly scroll down
> to see how my choices had affected my Assessment Score. So I've moved the
> score panel to the middle-right of the screen.
> The results are I can now see that my score can influence the Impact level.
> [image: image]
> <https://user-images.githubusercontent.com/20051/62815975-4fbb8300-bad5-11e9-8cb2-64612d846ead.png>
>
> After adding Health Related services
> [image: image]
> <https://user-images.githubusercontent.com/20051/62815986-77aae680-bad5-11e9-8ade-b5c3363268ee.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/canada-ca/aia-eia-js/pull/123
> Commit Summary
>
>    - Grammatical correction - "use data"
>    - Position Score where user easily can monitor it
>    - Revert "Position Score where user easily can monitor it"
>    - Position Score where user can easily monitor it
>
> File Changes
>
>    - *M* plainData.json
>    <https://github.com/canada-ca/aia-eia-js/pull/123/files#diff-0> (2)
>    - *M* src/components/Score.vue
>    <https://github.com/canada-ca/aia-eia-js/pull/123/files#diff-1> (14)
>    - *M* src/survey-enfr.json
>    <https://github.com/canada-ca/aia-eia-js/pull/123/files#diff-2> (2)
>
> Patch Links:
>
>    - https://github.com/canada-ca/aia-eia-js/pull/123.patch
>    - https://github.com/canada-ca/aia-eia-js/pull/123.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/canada-ca/aia-eia-js/pull/123?email_source=notifications&email_token=AAM4TZOY7OLUSWS6NRRV62DQDYMKHA5CNFSM4IKYIU4KYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HEQH2YA>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM4TZLQLLD3H4XCVVWMGYLQDYMKHANCNFSM4IKYIU4A>
> .
>

dfdumaresq commented:
Thanks, @gcharest!
The scores don't fit in the box when the locale is French. It's not pretty so I'm working on that now.
![image](https://user-images.githubusercontent.com/20051/62824607-77eac680-bb55-11e9-90b5-9f712da85843.png)

gcharest commented:
OK, will wait before testing.

Also was wondering, what's the render like on mobile?

dfdumaresq commented:
I think I've got this. A code review might be helpful though; this is my first time into Vue.

![image](https://user-images.githubusercontent.com/20051/62825631-fb131900-bb63-11e9-9ecf-49b6a4c6a728.png)

dfdumaresq commented:
> Also was wondering, what's the render like on mobile?

I'll check and let you know, but this will from my localhost.

dfdumaresq commented:
Ah yes. Visual Rendering (_not_ Rendering time... heh) It's not great, but better in Landscape mode:

![IMG_2656](https://user-images.githubusercontent.com/20051/62825838-d1a7bc80-bb66-11e9-8892-cebd2c97f704.PNG)

![IMG_2654](https://user-images.githubusercontent.com/20051/62825852-f56b0280-bb66-11e9-9e62-5d36a7f1c57f.PNG)

gcharest commented:
Thanks! Looking good in desktop mode in both locale!

OK, I'll check if there's a way to have the CSS behave differently in mobile in Vue. Not sure how, maybe @CalvinRodo or @e-wu would have an idea.

dfdumaresq commented:
I'm thinking of using a navbar from Boostrap-vue. 
Here's a mockup of where I'm going.
![image](https://user-images.githubusercontent.com/20051/62838133-f9f9ee80-bc2c-11e9-91b9-cf64b0218505.png)

gcharest commented:
Oh, that's pretty nice!

Le dim. 11 août 2019 14 h 42, dfdumaresq <notifications@github.com> a
écrit :

> I'm thinking of using a navbar from Boostrap-vue.
> Here's a mockup of where I'm going.
> [image: image]
> <https://user-images.githubusercontent.com/20051/62838133-f9f9ee80-bc2c-11e9-91b9-cf64b0218505.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/canada-ca/aia-eia-js/pull/123?email_source=notifications&email_token=AAM4TZMGYOGOFLHS3Q4RASTQEBMQLA5CNFSM4IKYIU4KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4BGKZI#issuecomment-520250725>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM4TZIFHZLRHOFF77P35QLQEBMQLANCNFSM4IKYIU4A>
> .
>

dfdumaresq commented:
I've implemented the above example. Styles could use a fair amount of tweaking.

![image](https://user-images.githubusercontent.com/20051/62839057-db4e2480-bc39-11e9-907c-a93c6805dd1f.png)

gcharest commented:
Cool! I don't have access to my computer today but will try to get to it this evening. Otherwise, I'm really not a designer so I'd probably try to touch base with @CalvinRodo who may know someone 🤔

dfdumaresq commented:
Bootstrap-vue allowed me to remove the locale checking. That's a win!

![image](https://user-images.githubusercontent.com/20051/62841579-a86a5780-bc5e-11e9-96fd-7883f903e499.png)

e-wu commented:
I like the bottom floating bar :) Nice job.

For later PRs that would be nice to have
- Can we add the "back" "forward" buttons as well? 
- Can the score be in initial "collapsed" state where it is maximum 1 liner, so when vertical resolution is small, the floating bar won't cover the viewport?

MrDeshaies commented:
Great contribution, thank you Dave! What follows are a few minor niggles on design.

Using the hamburger/menu icon for collapsing doesn't seem right; a downward arrow would seem more appropriate, or the "-" like in Windows.

On the last screenshot (desktop version in French), the box is titled "AIA SCORE" which is English. Maybe we could simply use "Results" (English) and "Résultats" (French).

Finally, the color (heh). I'm not a huge fan of the teal. I suggest a light grey to harmonize with the other styles.

dfdumaresq commented:
Thanks for the feedback guys! I’m going to be away from my computer for a few days, plus back to my day-job on Friday, so my productivity will scale back a bit. But I will address the issues you’re concerned with. 

A main concern is improving the dashboard for small phones in portrait mode so that it does not take up too much real estate. 
My challenge is positioning the four score categories. Is there any flexibility with the wording? 
I would like to try dropping the word score, which will give more room for the categories and values. I think the meaning will still be clear.

dfdumaresq commented:
Wondering what you think of using icons in the score container? Here's an example with acronyms, which we could change to icons if in agreement.

![Image-1](https://user-images.githubusercontent.com/20051/63216636-067abd00-c0ed-11e9-8838-7c061cb7c21c.jpg)

dfdumaresq commented:
1) I've kept a single line container and I've dropped the Teal colour. 
2) I've added acronyms for the result labels on mobile screens. These are translated for French.

gcharest commented:
That looks pretty good! What do you think @MrDeshaies ?

MrDeshaies commented:
Yeeeessss! Looking good! Great job @gcharest @dfdumaresq

gcharest commented:
All @dfdumaresq work! I will run a few accessibility tests this PM.

dfdumaresq commented:
@gcharest, I've compared master with my branch and can see the accessibility has degraded. Input fields are narrower, select options are in a smaller font size, for example. I suspect my introduction of bootstrap is the culprit.
```"bootstrap": "^4.3.1", "bootstrap-vue": "^2.0.0-rc.28",``

gcharest commented:
Thanks! Hmm, I'm think we might need to bring in @thriuin for input. Can we deploy this on open.canada.ca?

Who tests/vets the accesibility on your team? What would be your acceptance threshold?

Thanks!

dfdumaresq commented:
@gcharest , I've fixed up the css issues. Have a look:

Desktop
![aia-desktop-css-fix](https://user-images.githubusercontent.com/20051/64483068-78589a80-d1b1-11e9-9024-ccfe272e46f5.png)

Mobile - iPhone 6
![aia-mobile-css-fix](https://user-images.githubusercontent.com/20051/64483053-4c3d1980-d1b1-11e9-82c5-70111bbc6c7e.png)

gcharest commented:
Awesome! I think we can merge now. I'll contact our colleagues at
open.canada.ca to confirm if they need anything else before deploying as
well!

Le sam. 7 sept. 2019 23 h 53, dfdumaresq <notifications@github.com> a
écrit :

> @gcharest <https://github.com/gcharest> , I've fixed up the css issues.
> Have a look:
>
> Desktop
> [image: aia-desktop-css-fix]
> <https://user-images.githubusercontent.com/20051/64483068-78589a80-d1b1-11e9-9024-ccfe272e46f5.png>
>
> Mobile - iPhone 6
> [image: aia-mobile-css-fix]
> <https://user-images.githubusercontent.com/20051/64483053-4c3d1980-d1b1-11e9-82c5-70111bbc6c7e.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/canada-ca/aia-eia-js/pull/123?email_source=notifications&email_token=AAM4TZJUQXIV7PDQN52WN7DQIRZLFA5CNFSM4IKYIU4KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6FHGOA#issuecomment-529167160>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM4TZL7FWPNYONWZL7RZBLQIRZLFANCNFSM4IKYIU4A>
> .
>

thriuin commented:
Hi @gcharest, back from leave. For deployment, I would start with the staging site first for review.

gcharest commented:
Awesome, thanks for the feedback!

Le lun. 9 sept. 2019 10 h 46, Ross Thompson <notifications@github.com> a
écrit :

> Hi @gcharest <https://github.com/gcharest>, back from leave. For
> deployment, I would start with the staging site first for review.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/canada-ca/aia-eia-js/pull/123?email_source=notifications&email_token=AAM4TZJZUV67QQT2PODWXX3QIZOSZA5CNFSM4IKYIU4KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6H3ENY#issuecomment-529510967>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM4TZNK6Z3JTRUBAEMQPBLQIZOSZANCNFSM4IKYIU4A>
> .
>

gcharest commented:
Hi @dfdumaresq !

I was about to merge the PR but happened to be using Chrome for that and there seems to be an unexpected floating issue (I think?) Firefox doesn't create the same issue, the page width and its elements all fit right into place.

![image](https://user-images.githubusercontent.com/1690085/65285069-6825b100-db09-11e9-87c0-33cd9ded9811.png)

dfdumaresq commented:
I'll take a look at it @gcharest.

gcharest commented:
The last commit created the issue. I'm going over the CSS modifications to narrow down where it could have lead to this strange behaviour.

gcharest commented on public/survey.css:
```diff
@@ -1,3 +1,168 @@
+  margin-left: auto !important;
+}
+
+.h1, .h2, .h3, .h4, .h5, .h6, h1, h2, h3, h4, h5, h6 {
```
Issues seems wot bhe with the changes to headers somehow.

dfdumaresq replied:
Okay - I'm glad you've resolved the floating issue.

gcharest replied:
Was rushing to get the kids, haven't had the chance to fix it yet lol :grin:

dfdumaresq replied:
Ah - I have one too (he just turned 5 and started Kindergarten!) I thought you wanted me to fix this... So I'll get out of your way. Just let me know if you want me to do anything.

gcharest replied:
No, no, if you have a few cycles, you're way better than me at CSS just from looking at the magic you've done :grinning:

gcharest replied:
So, as far as I can tell, the issue is with h2, which seems to be used throught the [WET](https://www.canada.ca/en/treasury-board-secretariat/services/government-communications/web-experience-toolkit.html) template via the script for screen reader content.

By changing the behaviour of the CSS for our own h2 headers, it looks like it interferes with the injected HTML. At least that's what I think is happening :thinking:

gcharest replied:
My only workaround for now would be to remove h2 as an element to be included on line 32.

If you have another option, please go ahead. If not, if you could please fix and push your commit, I'll merge the PR and we'll have a new UI for the score! :grinning: 

Thanks so much for all the great work you've done on this project!

dfdumaresq replied:
Okay - I'm on it.

dfdumaresq replied:
The only problem is I can't reproduce what you're seeing. 
I'm using Chrome Version 77. This is what I get when I hit the 'Start Again' button.

![image](https://user-images.githubusercontent.com/20051/65366149-0cc0f500-dbd5-11e9-9ebb-0cf21179ba7b.png)

dfdumaresq replied:
Your image above seems to suggest you have scrollbars appearing in the window. Could you try Window|Zoom? See if that makes a difference? Also, of course clear your cache. You can also try a Empty Cache and Hard Reload, but you need the Inspector open to see that after holding down the reload button.

![image](https://user-images.githubusercontent.com/20051/65366184-907ae180-dbd5-11e9-99a2-c882826a7a93.png)

dfdumaresq replied:
Safari v13
![image](https://user-images.githubusercontent.com/20051/65366245-37f81400-dbd6-11e9-8566-dd043d2964c1.png)

Firefox v69
![image](https://user-images.githubusercontent.com/20051/65366258-45ad9980-dbd6-11e9-820e-04c5f418ca9e.png)

gcharest replied:
OK, testing it out!

gcharest replied:
Hmm, so I tested on Chrome 77, both on Windows 10 Pro and Ubuntu 18.04 and both behave the same. The bottom scroll bar is there for Chrome but not other browsers 🤔 

Behaviour is as expected with Edge 44 and Firefox 69.0.1. This is really odd... Will try on a MacOS device to see what's going on.

gcharest replied:
![image](https://user-images.githubusercontent.com/1690085/65374797-0590fa00-dc5c-11e9-9c77-fba79db17d94.png)

dfdumaresq replied:
I can now reproduce the scrollbars you mention in Chrome, I think. But tell me are you able to scroll to the left and show the page correctly?

dfdumaresq replied:
I've followed your suggestion and it works on my Mac.

![image](https://user-images.githubusercontent.com/20051/65380786-b072d980-dc97-11e9-9bae-f8dbaac47ee7.png)

gcharest commented:
Thanks so much! This is really great work!

dfdumaresq commented:
Excellent, @gcharest! That was fun!
I've found another grammar correction and I'll look for some other issues I can tackle. Let me know if there's anything in particular that you'd like me to look at.
